### ISSUE TITLE
Fix error in detecting config

### ISSUE THREAD

**Issue by micahjsmith** [ballet/ballet-bot#25]
Stacktrace
```
TypeError: Cannot read property 'github' of undefined
  File "/app/index.js", line 163, col 21, in shouldPruneRedundantFeatures
    } else if (config.github.pruning_action === 'no_action') {
  File "async /app/index.js", line 59, col 31, in null.<anonymous>
```

Event:
```
{"event":"check_run.completed","id":"d346e5d0-9250-11eb-8e37-3fd1bb21cccd","installation":5204583,"repository":"HDI-Project/ballet-predict-census-income"}
```

From sentry issue 2309327746

**Comment by micahjsmith** [comment]
The full logs from redelivering the webhook

<details>
2021-04-02T01:58:14.406230+00:00 app[web.1]: 01:58:14.401Z TRACE http: POST / (id=51da7fab-55d1-45e4-b2df-bff3e6c9e6d4, req.remoteAddress=::ffff:10.63.70.171, req.remotePort=13249)
2021-04-02T01:58:14.406245+00:00 app[web.1]: POST / HTTP/1.1
2021-04-02T01:58:14.406246+00:00 app[web.1]: host: ballet-bot.herokuapp.com
2021-04-02T01:58:14.406247+00:00 app[web.1]: connection: close
2021-04-02T01:58:14.406247+00:00 app[web.1]: user-agent: GitHub-Hookshot/89c2940
2021-04-02T01:58:14.406248+00:00 app[web.1]: accept: */*
2021-04-02T01:58:14.406250+00:00 app[web.1]: x-github-delivery: d346e5d0-9250-11eb-8e37-3fd1bb21cccd
2021-04-02T01:58:14.406250+00:00 app[web.1]: x-github-event: check_run
2021-04-02T01:58:14.406250+00:00 app[web.1]: x-github-hook-id: 158082350
2021-04-02T01:58:14.406251+00:00 app[web.1]: x-github-hook-installation-target-id: 46883
2021-04-02T01:58:14.406251+00:00 app[web.1]: x-github-hook-installation-target-type: integration
2021-04-02T01:58:14.406253+00:00 app[web.1]: x-hub-signature: sha1=1a60bbb8eeb50b0f104505be28770fbc0365730b
2021-04-02T01:58:14.406254+00:00 app[web.1]: x-hub-signature-256: sha256=b7b938e78c258f3d62ac8bfb86e5b4df109a2e8e7706e7abfaae8324d7dc3b98
2021-04-02T01:58:14.406254+00:00 app[web.1]: content-type: application/json
2021-04-02T01:58:14.406255+00:00 app[web.1]: x-request-id: 51da7fab-55d1-45e4-b2df-bff3e6c9e6d4
2021-04-02T01:58:14.406255+00:00 app[web.1]: x-forwarded-for: 140.82.115.244
2021-04-02T01:58:14.406255+00:00 app[web.1]: x-forwarded-proto: https
2021-04-02T01:58:14.406256+00:00 app[web.1]: x-forwarded-port: 443
2021-04-02T01:58:14.406256+00:00 app[web.1]: via: 1.1 vegur
2021-04-02T01:58:14.406257+00:00 app[web.1]: connect-time: 0
2021-04-02T01:58:14.406257+00:00 app[web.1]: x-request-start: 1617328694400
2021-04-02T01:58:14.406258+00:00 app[web.1]: total-route-time: 0
2021-04-02T01:58:14.406258+00:00 app[web.1]: content-length: 16162
2021-04-02T01:58:14.558695+00:00 app[web.1]: 01:58:14.558Z DEBUG probot: Webhook received
2021-04-02T01:58:14.558698+00:00 app[web.1]: event: {
2021-04-02T01:58:14.558698+00:00 app[web.1]: "event": "check_run.completed",
2021-04-02T01:58:14.558701+00:00 app[web.1]: "id": "d346e5d0-9250-11eb-8e37-3fd1bb21cccd",
2021-04-02T01:58:14.558702+00:00 app[web.1]: "installation": 5204583,
2021-04-02T01:58:14.558702+00:00 app[web.1]: "repository": "HDI-Project/ballet-predict-census-income"
2021-04-02T01:58:14.558702+00:00 app[web.1]: }
2021-04-02T01:58:14.564959+00:00 app[web.1]: 01:58:14.564Z  INFO event: Responding to check_run (id=d346e5d0-9250-11eb-8e37-3fd1bb21cccd) (id=d346e5d0-9250-11eb-8e37-3fd1bb21cccd)
2021-04-02T01:58:16.535071+00:00 app[web.1]: 01:58:16.534Z DEBUG github: GitHub request: GET /repos/:owner/:repo/contents/:path - undefined (installation=5204583)
2021-04-02T01:58:16.535087+00:00 app[web.1]: params: {
2021-04-02T01:58:16.535088+00:00 app[web.1]: "baseUrl": "https://api.github.com",
2021-04-02T01:58:16.535090+00:00 app[web.1]: "mediaType": {
2021-04-02T01:58:16.535090+00:00 app[web.1]: "format": "",
2021-04-02T01:58:16.535091+00:00 app[web.1]: "previews": []
2021-04-02T01:58:16.535091+00:00 app[web.1]: },
2021-04-02T01:58:16.535091+00:00 app[web.1]: "request": {
2021-04-02T01:58:16.535092+00:00 app[web.1]: "validate": {
2021-04-02T01:58:16.535092+00:00 app[web.1]: "owner": {
2021-04-02T01:58:16.535093+00:00 app[web.1]: "required": true,
2021-04-02T01:58:16.535093+00:00 app[web.1]: "type": "string"
2021-04-02T01:58:16.535093+00:00 app[web.1]: },
2021-04-02T01:58:16.535094+00:00 app[web.1]: "path": {
2021-04-02T01:58:16.535094+00:00 app[web.1]: "required": true,
2021-04-02T01:58:16.535095+00:00 app[web.1]: "type": "string"
2021-04-02T01:58:16.535095+00:00 app[web.1]: },
2021-04-02T01:58:16.535095+00:00 app[web.1]: "ref": {
2021-04-02T01:58:16.535096+00:00 app[web.1]: "type": "string"
2021-04-02T01:58:16.535096+00:00 app[web.1]: },
2021-04-02T01:58:16.535096+00:00 app[web.1]: "repo": {
2021-04-02T01:58:16.535097+00:00 app[web.1]: "required": true,
2021-04-02T01:58:16.535097+00:00 app[web.1]: "type": "string"
2021-04-02T01:58:16.535097+00:00 app[web.1]: }
2021-04-02T01:58:16.535098+00:00 app[web.1]: }
2021-04-02T01:58:16.535098+00:00 app[web.1]: },
2021-04-02T01:58:16.535098+00:00 app[web.1]: "owner": "HDI-Project",
2021-04-02T01:58:16.535099+00:00 app[web.1]: "repo": "ballet-predict-census-income",
2021-04-02T01:58:16.535099+00:00 app[web.1]: "path": "ballet.yml"
2021-04-02T01:58:16.535099+00:00 app[web.1]: }
2021-04-02T01:58:16.736555+00:00 app[web.1]: 01:58:16.736Z  INFO event: Getting a check from branch: master (id=d346e5d0-9250-11eb-8e37-3fd1bb21cccd)
2021-04-02T01:58:16.742643+00:00 app[web.1]: 01:58:16.742Z  INFO event: On commit: Merge branch 'project-template' (id=d346e5d0-9250-11eb-8e37-3fd1bb21cccd)
2021-04-02T01:58:16.876296+00:00 app[web.1]: 01:58:16.875Z DEBUG github: GitHub request: GET /repos/:owner/:repo/git/commits/:commit_sha - undefined (installation=5204583)
2021-04-02T01:58:16.876321+00:00 app[web.1]: params: {
2021-04-02T01:58:16.876322+00:00 app[web.1]: "baseUrl": "https://api.github.com",
2021-04-02T01:58:16.876323+00:00 app[web.1]: "mediaType": {
2021-04-02T01:58:16.876323+00:00 app[web.1]: "format": "",
2021-04-02T01:58:16.876323+00:00 app[web.1]: "previews": []
2021-04-02T01:58:16.876324+00:00 app[web.1]: },
2021-04-02T01:58:16.876324+00:00 app[web.1]: "request": {
2021-04-02T01:58:16.876325+00:00 app[web.1]: "validate": {
2021-04-02T01:58:16.876325+00:00 app[web.1]: "commit_sha": {
2021-04-02T01:58:16.876325+00:00 app[web.1]: "required": true,
2021-04-02T01:58:16.876326+00:00 app[web.1]: "type": "string"
2021-04-02T01:58:16.876326+00:00 app[web.1]: },
2021-04-02T01:58:16.876327+00:00 app[web.1]: "owner": {
2021-04-02T01:58:16.876327+00:00 app[web.1]: "required": true,
2021-04-02T01:58:16.876327+00:00 app[web.1]: "type": "string"
2021-04-02T01:58:16.876328+00:00 app[web.1]: },
2021-04-02T01:58:16.876328+00:00 app[web.1]: "repo": {
2021-04-02T01:58:16.876328+00:00 app[web.1]: "required": true,
2021-04-02T01:58:16.876329+00:00 app[web.1]: "type": "string"
2021-04-02T01:58:16.876329+00:00 app[web.1]: }
2021-04-02T01:58:16.876329+00:00 app[web.1]: }
2021-04-02T01:58:16.876329+00:00 app[web.1]: },
2021-04-02T01:58:16.876330+00:00 app[web.1]: "owner": "HDI-Project",
2021-04-02T01:58:16.876330+00:00 app[web.1]: "repo": "ballet-predict-census-income",
2021-04-02T01:58:16.876331+00:00 app[web.1]: "commit_sha": "ed7322617fa13623573b297e1f00164c5736c4a7"
2021-04-02T01:58:16.876331+00:00 app[web.1]: }
2021-04-02T01:58:16.999548+00:00 app[web.1]: 01:58:16.999Z ERROR event: Cannot read property 'github' of undefined (id=d346e5d0-9250-11eb-8e37-3fd1bb21cccd)
2021-04-02T01:58:16.999550+00:00 app[web.1]: TypeError: Cannot read property 'github' of undefined
2021-04-02T01:58:16.999551+00:00 app[web.1]: at shouldPruneRedundantFeatures (/app/index.js:163:21)
2021-04-02T01:58:16.999551+00:00 app[web.1]: at async /app/index.js:59:31
2021-04-02T01:58:16.999552+00:00 app[web.1]: --
2021-04-02T01:58:16.999553+00:00 app[web.1]: event: {
2021-04-02T01:58:16.999553+00:00 app[web.1]: "event": "check_run.completed",
2021-04-02T01:58:16.999554+00:00 app[web.1]: "id": "d346e5d0-9250-11eb-8e37-3fd1bb21cccd",
2021-04-02T01:58:16.999559+00:00 app[web.1]: "installation": 5204583,
2021-04-02T01:58:16.999560+00:00 app[web.1]: "repository": "HDI-Project/ballet-predict-census-income"
2021-04-02T01:58:16.999560+00:00 app[web.1]: }
2021-04-02T01:58:17.011038+00:00 app[web.1]: 01:58:17.010Z ERROR probot: Cannot read property 'github' of undefined
2021-04-02T01:58:17.011041+00:00 app[web.1]: TypeError: Cannot read property 'github' of undefined
2021-04-02T01:58:17.011041+00:00 app[web.1]: at shouldPruneRedundantFeatures (/app/index.js:163:21)
2021-04-02T01:58:17.011042+00:00 app[web.1]: at async /app/index.js:59:31
2021-04-02T01:58:17.012951+00:00 app[web.1]: 01:58:17.012Z  INFO http: POST / 500 - 2606.40 ms (id=51da7fab-55d1-45e4-b2df-bff3e6c9e6d4)
2021-04-02T01:58:17.013538+00:00 app[web.1]: 01:58:17.013Z TRACE http:  (id=51da7fab-55d1-45e4-b2df-bff3e6c9e6d4, res.duration=2606.40)
2021-04-02T01:58:17.013540+00:00 app[web.1]: HTTP/1.1 500 Internal Server Error
2021-04-02T01:58:17.013541+00:00 app[web.1]: x-powered-by: Express
2021-04-02T01:58:17.013542+00:00 app[web.1]: x-request-id: 51da7fab-55d1-45e4-b2df-bff3e6c9e6d4
2021-04-02T01:58:17.015450+00:00 heroku[router]: at=info method=POST path="/" host=ballet-bot.herokuapp.com request_id=51da7fab-55d1-45e4-b2df-bff3e6c9e6d4 fwd="140.82.115.244" dyno=web.1 connect=0ms service=2613ms status=500 bytes=217 protocol=https
2021-04-02T01:58:17.482906+00:00 app[web.1]: (node:30) [DEP0066] DeprecationWarning: OutgoingMessage.prototype._headers is deprecated
2021-04-02T01:58:17.482934+00:00 app[web.1]: (Use `node --trace-deprecation ...` to show where the warning was created)
</details>

**Comment by micahjsmith** [comment]
Fixed in 0a5c3e9
